frontend/src/helpers/transformations.js

```diff
@@ -115,3 +103,25 @@ export const recursivelyProcessTree = (treeContent, treeType, level, id, parent)
 
   return processedTree;
 };
+
+// Converts org chart trees into an address book (flattened array of strings).
+export const recursivelyCreateAddressBook = (treeContent, treeType) => {
+  let processedTree = [];
+  // For each node in the tree, process the node to the expected output for the tree view.
+  for (let i = 0; i < treeContent.length; i++) {
+    const treeNode = treeContent[i];
+
+    // If this node has children.
+    if (treeNode[treeType]) {
+      // Update the processed set of the tree.
+      processedTree.push(treeNode.text);
```

Comment:
Since this line is both in the if and the else, it feels like it should go before the if; delete the else; and maybe comment that you don't need to do anything else if it is a leaf node.